pkg/config/config_parser.go

```diff
@@ -32,66 +39,64 @@ func (p *Parser) LoadConfigFile(path string) (*Config, diag.Diagnostics) {
 	return p.LoadConfigFromSource(contents)
 }
 
-func ValidateCQBlock(cq *CloudQuery) diag.Diagnostics {
+// ProcessConfig handles the configuration after it was loaded and parsed
+// 1. Assigns defaults after decoding the raw configuration format
+// 2. Overrides configuration values from CLI flags
+// 3. Validates the configuration provided by the user
+// 4. Normalizes the configuration to make it easier to use
+func ProcessConfig(config *Config) diag.Diagnostics {
+	assignDefaults(config)
+	overrideFromCLIFlags(config)
+	diags := validate(config)
+	if diags.HasErrors() {
+		return diags
+	}
+
+	normalize(config)
+	return diags
+}
+
+func validate(config *Config) diag.Diagnostics {
 	var diags diag.Diagnostics
 
-	if cq.Connection == nil {
-		cq.Connection = &Connection{}
-	}
+	diags = diags.Add(validateCloudQueryProviders(config.CloudQuery.Providers))
+	diags = diags.Add(validateConnection(config.CloudQuery.Connection))
+	diags = diags.Add(validateProvidersBlock(config))
+
+	return diags
```

Comment:
```suggestion
	return diags.Add(validateProvidersBlock(config))
```

Reply:
@roneli how about
```go
return diag.Diagnostics{}.Add(validateCloudQueryProviders(config.CloudQuery.Providers)).Add(validateConnection(config.CloudQuery.Connection)).Add(validateProvidersBlock(config))
```
😝

Reply:
Ah, I did that intentionally to make it easier to add more validations functions before/after each function, but doesn't really matter now

Reply:
It's all the same to me expect that having it line by line makes it easier to debug